src/Libraries/CoreNodesUI/Input/DoubleSlider.cs

```diff
@@ -117,9 +99,54 @@ protected override bool UpdateValueCore(string name, string value, UndoRedoRecor
                         this.Min = Value;
                     }
                     return true; // UpdateValueCore handled.
+                case "Step":
+                case "StepText":
+                    Step = SliderViewModel<double>.ConvertStringToDouble(value);
+                    return true;
             }
 
             return base.UpdateValueCore(name, value, recorder);
         }
+
+        #region Serialization/Deserialization Methods
+
+        protected override void SerializeCore(XmlElement element, SaveContext context)
+        {
+            base.SerializeCore(element, context); // Base implementation must be called.
+
+            var xmlDocument = element.OwnerDocument;
+            XmlElement subNode = xmlDocument.CreateElement("Range");
+            subNode.SetAttribute("min", Min.ToString(CultureInfo.InvariantCulture));
+            subNode.SetAttribute("max", Max.ToString(CultureInfo.InvariantCulture));
+            subNode.SetAttribute("step", Step.ToString(CultureInfo.InvariantCulture));
+            element.AppendChild(subNode);
+        }
+
+        protected override void DeserializeCore(XmlElement element, SaveContext context)
+        {
+            base.DeserializeCore(element, context); //Base implementation must be called.
+
+            foreach (XmlNode subNode in element.ChildNodes)
+            {
+                if (!subNode.Name.Equals("Range"))
+                    continue;
+                if (subNode.Attributes == null || (subNode.Attributes.Count <= 0))
+                    continue;
+
+                foreach (XmlAttribute attr in subNode.Attributes)
+                {
+                    if (attr.Name.Equals("min"))
+                        Min = Convert.ToDouble(attr.Value, CultureInfo.InvariantCulture);
+                    else if (attr.Name.Equals("max"))
+                        Max = Convert.ToDouble(attr.Value, CultureInfo.InvariantCulture);
+                    else if (attr.Name.Equals("step"))
+                        Step = Convert.ToDouble(attr.Value, CultureInfo.InvariantCulture);
+                }
```

Comment:
Consider using a switch instead, that has a default case that logs that its seen data it can't de-serialise. This should probably be a crash in a debug build?